CRM/Contact/BAO/Query.php

```diff
@@ -1303,6 +1303,9 @@ public function addHierarchicalElements() {
                   break;
 
                 default:
+                  if (substr($tableName, 0, 14) == 'civicrm_value_') {
```

Comment:
It is possible to rename custom group tables - if you look at the fieldSpec (I think the var is addressCustomFields) the presence of a 'custom_group_id' should be indicative though

I kinda like being even more clear & adding is_custom_field to that spec for custom fields (in BAO_CustomField::getFields) but custom_group_id should be reliable

Reply:
Here, now it this looks like it's on par with what you mentioned on the fieldSpec of addressCustomFields.